classes/Log/LegacyLogger.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/*
+ * 2007-2018 PrestaShop
+ * 
+ * NOTICE OF LICENSE
+ * 
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ * 
+ * DISCLAIMER
+ * 
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ * 
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2018 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Log;
+
+/**
+ * This class reimplement the old properties of the class AdminSelfUpgrade,
+ * where all the mesages were stored
+ */
+class LegacyLogger extends Logger
+{
+    protected $normalMessages = array();
+    protected $severeMessages = array();
+    protected $lastInfo = '';
+
+    /**
+     * @var Resource File descriptor of the log file
+     */
+    protected $fd = null;
+
+    public function __constuct($fileName = null)
+    {
+        if (!is_null($fileName)) {
```

Comment:
!== is faster than is_null 
FYI: https://stackoverflow.com/questions/8228837/is-nullx-vs-x-null-in-php
There are many bench about why is_null is slow =)